### ISSUE TITLE
Endpoint-specific schema transform

### ISSUE THREAD

**Issue by thorhj** [fastify/fastify-swagger#749]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

fastify-swagger allows choose a [transform function](https://github.com/fastify/fastify-swagger#register.options.transform) in case you have a different schema type than the built-in one.

You can have different schema types for different endpoints, but to my knowledge fastify-swagger does not support this (I couldn't find anything for this at least). I would like to have an endpoint-specific transform function that differs from the "global" transform function. Something like this is what I imagine:

```js
fastify.get("/my-endpoint", {
  schema: { ... },
  config: {
    swaggerTransform: ( ... ) => { ... }
  }
},
async (request, reply) => { ... });
```

This `swaggerTransform` property should be able to indicate the "default" (no) transform or any other transform function.

### Motivation

In some of my projects I am [fastify-type-provider-zod](https://www.npmjs.com/package/fastify-type-provider-zod) which has its own transform function for this use-case.

However, I have found a need for having a specific endpoint NOT using zod schemas to define the schema. In my case, we would like to use a plugin that adds a new route-shorthand to fastify to easily create AWS SNS event webhooks correctly, and in this plugin the schema is implemented using the default schema declaration. When mixing schema types in the project, fastify-swagger will fail because it expects the routes to have zod schemas, but this one endpoint does not.

My current workaround is to hide the endpoint in question from swagger, but I would like to have it included, and to do that I think I need an endpoint-specific transform function for the endpoints in question.

**Comment by Uzlopak** [comment]
Does swagger really expect zod schemas? 

@kibertoad

**Comment by kibertoad** [comment]
@Uzlopak not swagger per se, but if type-provider-zod compiler and validator are registered, that will be the format of schemas that they will expect

**Comment by kibertoad** [comment]
see https://github.com/turkerdev/fastify-type-provider-zod/blob/85675fad44c7679622d80fb1925f5901cc2f57f8/src/index.ts#L32

**Comment by thorhj** [comment]
By the way, if you think this is a good idea and if we make a design for the interface (there are probably opinions on my suggestion), I could work on this feature myself.

**Comment by MatanYadaev** [comment]
@mcollina Suggested this API: https://github.com/fastify/fastify-swagger/issues/727#issuecomment-1618258696

**Comment by fredrikj31** [comment]
I really need this feature as well, for a project using zod type provider, as a alternative schema parser.

Is this being shipped in the next version, or what's the status of it? 🤔

**Comment by Uzlopak** [comment]
@fredrikj31 
If you could test #752 and check if it solves your issue, I could easier approve. Personally I dont have a clue because I dont use zod etc..

**Comment by fredrikj31** [comment]
I have tried pulling it down, and using it on a fastify server with zod type provider, and it seems to be working, so it looks like it solves my problem. Can this be released so I can use it on my apis?

**Comment by Uzlopak** [comment]
I reviewed it now. Waiting for feedback by @thorhj

**Comment by fredrikj31** [comment]
Okay, super. Thanks for the status update. Will be watching the PR 🙏

**Comment by mcollina** [comment]
shipped v8.10.0
